deepchecks/base/dataset.py

```diff
@@ -420,6 +441,20 @@ def train_test_split(self,
                                              stratify=stratify)
         return self.copy(train_df), self.copy(test_df)
 
+    @staticmethod
+    def _infer_label_type(
+            label_col: pd.Series,
+            max_categorical_ratio: float,
+            max_categories: int,
+            max_float_categories: int
+    ):
+        if not is_numeric_dtype(label_col):
+            return 'classification label'
+        elif is_categorical(label_col, max_categorical_ratio, max_categories, max_float_categories):
+            return 'classification label'
+        else:
+            return 'regression label'
```

Comment:
I just hope it's a good heuristic....

Reply:
I'm also not sure about it, but it's better than using something like https://scikit-learn.org/stable/modules/generated/sklearn.utils.multiclass.type_of_target.html#sklearn.utils.multiclass.type_of_target - this will for e.g. infer a "test score" that is ints from 0 to 100 as "classification" although it's clearly regression. 

We're open to suggestions